### PR TITLE
Feature/unplan assignment

### PR DESCRIPTION
**Pull request by TimvdLippe**

### CODE REVIEW

File: src/main/java/nl/tudelft/planningstool/api/v1/UserOccurrenceAPI.java [hunk 1]
```diff
@@ -167,6 +168,18 @@ public UserOccurrenceResponse update(@PathParam("userId") String userId,
         return createListResponse(occurrences);
     }
 
+    @DELETE
+    @Path("/courses/COURSE-{courseId: (\\d|\\w|-)+}/{assignmentId: (\\d)+}")
+    public void deleteOccurrence(@PathParam("userId") String userId, @PathParam("courseId") String courseId, @PathParam("assignmentId") Integer assignmentId) {
+        final Course course = this.courseDAO.getFromUUID(courseId);
+        final User user = this.userDAO.getFromUUID(userId);
+        final Predicate<UserOccurrence> predicate = o -> o.getAssignment().getCourse().equals(course) && o.getAssignment().getId().intValue() == assignmentId.intValue();
```
Comment (jwgmeligmeyling) [inline src/main/java/nl/tudelft/planningstool/api/v1/UserOccurrenceAPI.java]: `o.getAssignment().getId().equals(assignmentId)` would be a little less verbose here :wink:

File: src/main/java/nl/tudelft/planningstool/database/entities/User.java [hunk 2]
```diff
@@ -72,7 +72,8 @@
     @OneToMany(mappedBy = "user")
     private Set<CourseRelation> courses = Sets.newHashSet();
 
-    @OneToMany(mappedBy = "user", cascade = CascadeType.ALL)
+    // http://stackoverflow.com/questions/549961/hibernate-removing-item-from-a-list-does-not-persist#comment12057916_550441
+    @OneToMany(mappedBy = "user", cascade = CascadeType.ALL, orphanRemoval = true)
```
Comment (jwgmeligmeyling) [inline src/main/java/nl/tudelft/planningstool/database/entities/User.java:76]: :+1: This is indeed the way to map Objects that cannot live without a parent object. `CascadeType.ALL` only ensures all `UserOccurrences` belonging to a specific user are removed then a `User` is removed.

**Comment by jwgmeligmeyling** [comment]
Few very minor remarks, but no issues. Looks good! :+1:

**Comment by GijsWeterings** [comment]
Does the removal progress also delete the original assignment object? The task list in the frontend doesn't find the unplanned assignment anymore

**Comment by TimvdLippe** [comment]
No it does not. You probably need to update the frontend branch, as the last commit fixes the refreshing of the assignment list. It was working fine for me

**Comment by GijsWeterings** [comment]
intellij pls, y u no autosync :grumble:

:shipit:

**Pull request merged**